packages/utils/src/types/index.ts

```diff
@@ -17,41 +17,18 @@ export type PackedGroth16Proof = [
 ]
 
 /**
- * Supported proof types:
- * {@link @zk-kit/eddsa-proof!generate | EdDSA}
- * {@link @zk-kit/poseidon-proof!generate | Poseidon}
- * {@link https://github.com/semaphore-protocol/semaphore/tree/main/packages/proof | Semaphore}.
- * @enum
- */
-export enum Proof {
-    EDDSA = "eddsa",
-    POSEIDON = "poseidon",
-    SEMAPHORE = "semaphore"
-}
-
-/**
- * Circom Snark Artifact file extensions.
- * @enum
- */
-export enum Artifact {
-    WASM = "wasm",
-    ZKEY = "zkey"
-}
-
-/**
- * @prop Artifact.WASM
- * @prop Artifact.ZKEY
+ * @prop SnarkArtifacts.wasm
+ * @prop SnarkArtifacts.zkey
  * @interface
  */
-export type SnarkArtifacts = Record<Artifact, string>
-
-type Digit = "0" | "1" | "2" | "3" | "4" | "5" | "6" | "7" | "8" | "9"
+export type SnarkArtifacts = Record<"wasm" | "zkey", string>
 
 /**
  * Semantic version.
  * @example
- * 1.0.0
+ * 1.0.0-beta
+ * 2.0.0
  * @example
  * "latest"
  */
-export type Version = `${Digit}.${Digit}.${Digit}` | "latest"
+export type Version = string | "latest"
```

Comment:
perfect typings is not the goals, but what about

```ts
type Digit = `${number}`;
type preRelease = "alpha" | "beta";

export type Version = 
  `${Digit}.${Digit}.${Digit}` | 
  `${Digit}.${Digit}.${Digit}-${string}` |
  `${Digit}.${Digit}.${Digit}-${preRelease}.${Digit}` |
  "latest";
```

Reply:
This should work 👍🏽

Reply:
```ts
`${Digit}.${Digit}.${Digit}-${preRelease}.${Digit}` |
```
This doesn't really work as the previous type includes all combinations.